frontend/lib/src/components/widgets/DateInput/DateInput.tsx

```diff
@@ -394,7 +394,7 @@ class DateInput extends React.PureComponent<Props, State> {
                             // The width property only enlarges its bounding box, so it's easier to click.
                             transform: "scale(1.41)",
                             width: theme.spacing.twoXL,
-                            marginRight: "-8px",
+                            marginRight: `calc(-${theme.spacing.sm})`,
```

Comment:
What is the difference between using `calc` and [this](https://github.com/streamlit/streamlit/blob/feature/button_group_segments/frontend/lib/src/components/shared/BaseButton/styled-components.ts#L348). While my example seems to work, I wonder if there's not the best practice.

Reply:
Ahh great callout, `calc` is likely overhead here just to put in the `-` sign 🙂 The code has been updated since your comment to not use a `-` sign anymore, but otherwise I would have used your linked approach